backend/core/src/shared/dto/pagination.dto.ts

```diff
@@ -64,3 +64,41 @@ export class PaginationQueryParams {
   })
   limit?: number
 }
+
+export class PaginationAllowsAllQueryParams {
+  @Expose()
+  @ApiPropertyOptional({
+    type: Number,
+    example: 1,
+    required: false,
+    default: 1,
+  })
+  @IsOptional({ groups: [ValidationsGroupsEnum.default] })
+  @IsNumber({}, { groups: [ValidationsGroupsEnum.default] })
+  @Transform((value: string | undefined) => (value ? parseInt(value) : 1), {
+    toClassOnly: true,
+  })
+  page?: number
+
+  @Expose()
+  @ApiPropertyOptional({
+    type: "number | 'all'",
+    example: 10,
+    required: false,
+    default: 10,
```

Comment:
Alternatively, we could default to all if no limit is specified.

Reply:
I prefer this over defaulting to all.